lib/chef/resource/powershell_package_source.rb

```diff
@@ -24,36 +25,131 @@ class PowershellPackageSource < Chef::Resource
+
+        ```ruby
+        powershell_package_source 'PowerShellModules -> GoldFishBowl' do
+          source_name                  "PowerShellModules"
+          new_name                     "GoldFishBowl"
+          provider_name                "Nuget"
+          trusted                      true
+          action                       :set
+        end
+        ```
+
+        **Update a Nuget package source with a new name when the source is secured with a username and password**:
+
+        ```ruby
+        powershell_package_source 'PowerShellModules -> GoldFishBowl' do
+          source_name                  "PowerShellModules"
+          new_name                     "GoldFishBowl"
+          trusted                      true
+          user_name                    "user@domain.io"
+          user_pass                    "some_secret_password"
+          action                       :set
+        end
+        ```
+
+        **Unregister a package source**:
+
+        ```ruby
+        powershell_package_source 'PowerShellModules' do
+          source_name                  "PowerShellModules"
+          action                       :unregister
+        end
+        ```
+      DOC
 
       property :source_name, String,
-        description: "The name of the package source.",
+        description: "A label that names your package source.",
         name_property: true
 
-      property :url, String,
-        description: "The URL to the package source.",
-        required: [:register]
+      property :new_name, String,
+        description: "Used when updating the name of a NON-PSRepository"
 
-      property :trusted, [TrueClass, FalseClass],
-        description: "Whether or not to trust packages from this source.",
-        default: false
+      property :source_location, String,
+        description: "The URL to the location to retrieve modules from."
 
-      property :provider_name, String,
-        equal_to: %w{ Programs msi NuGet msu PowerShellGet psl chocolatey },
-        validation_message: "The following providers are supported: 'Programs', 'msi', 'NuGet', 'msu', 'PowerShellGet', 'psl' or 'chocolatey'",
-        description: "The package management provider for the source.",
-        default: "NuGet"
+      alias :url :source_location
 
       property :publish_location, String,
-        description: "The URL where modules will be published to for this source. Only valid if the provider is `PowerShellGet`."
+        description: "The URL where modules will be published to. Only valid if the provider is `PowerShellGet`."
 
       property :script_source_location, String,
         description: "The URL where scripts are located for this source. Only valid if the provider is `PowerShellGet`."
 
       property :script_publish_location, String,
         description: "The location where scripts will be published to for this source. Only valid if the provider is `PowerShellGet`."
 
+      property :trusted, [TrueClass, FalseClass],
```

Comment:
These all need introduced lines for generating docs

Reply:
Done